src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -2091,6 +2094,73 @@ void GraphOptimizer::FuseEltwiseAndSimple(Graph &graph) {
     }
 }
 
+void GraphOptimizer::ShareReorders(Graph& graph) {
+    auto getSuitableReorder = [](NodePtr node) -> Reorder* {
+        if (node->getType() != Type::Reorder)
+            return nullptr;
+
+        Reorder* reorder = dynamic_cast<Reorder*>(node.get());
+        if (reorder == nullptr)
+            OPENVINO_THROW("Cannot get reorder layer ", node->getName());
+
+        if (!reorder->getShareable())
+            return nullptr;
+
+        // inplace children may write to data, cannot be safely shared
+        auto reorderConsumers = reorder->getChildEdgesAtPort(0);
```

Comment:
So, we have two main cases where we insert reorders into the graph generically:
1) Layout reorders
2) Inpace conflicts

From my understanding this optimization covers only Layout reorders. Is it correct?
If so, can we have a following pipeline, instead of checking for inplace case every time:
1) Insert Layout reorders
2) Share reorders
3) Insert Inplace-conflict reorders.

Reply:
Oh, sorry, my concern is invalid, you are correct!
Sharing reorders w/o considering inplace is good idea, since afterward ResolveInplaceConflicts() will handle inplace conflicts, thanks for your explanations! code is updated, please review again!